### PR TITLE
Update memory_statistics to use config file

### PR DESCRIPTION
**Pull request by archigup**
Since the library configuration is taken on the command line, it has to
be duplicated at all the call sites of the action. This has the action
get the library config from a config file so that multiple workflows can use the
same config. This will also aid the action's use in the hub repos.

### CODE REVIEW

File: memory_statistics/action.yml [hunk 1]
```diff
@@ -5,19 +5,9 @@ inputs:
     description: 'Path to library directory.'
     required: false
     default: ./
-  lib_name:
-    description: 'Library name for table header.'
+  config:
+    description: 'Configuration json file for memory estimation.'
```
Comment (aggarw13) [inline memory_statistics/action.yml:9]:
```suggestion
    description: 'JSON config file containing library information to be used for memory estimation.'
```

File: memory_statistics/memory_statistics.py [hunk 2]
```diff
@@ -38,17 +40,13 @@ def convert_size_to_kb(byte_size):
...
-    The output of size is expected to be the Berkley output format:
-    text       data     bss     dec     hex filename
-
-    values is an input defaultdict which maps filenames to dicts of opt level
-    to sizes.
-    This function adds each file's size to the file's dict with the key
-    provided by the key parameter.
-    '''
+    # output expects the output of the makefile, which ends in a call to
+    # arm-none-eabi-size The output of size is expected to be the Berkley
```
Comment (aggarw13) [inline memory_statistics/memory_statistics.py:44]: Why was this changed? The earlier formatting seemed more readable

**Review by aggarw13 — APPROVED** [review]

File: memory_statistics/action.yml [hunk 3]
```diff
@@ -30,3 +34,15 @@ runs:
           --config "${{ inputs.config }}"
           --output "${{ inputs.output }}"
       shell: bash
+    - name: Compare table
+      run: |
+          if [ "x${{ inputs.check_against }}" != "x" ] ; then
```
Comment (aggarw13) [inline memory_statistics/action.yml:39]: Does this not work to check if the input variable is empty?
`[ -z "${{ inputs.check_against }}" ]`

File: memory_statistics/action.yml [hunk 4]
```diff
@@ -5,23 +5,17 @@ inputs:
...
-  compiler_flags:
-    description: 'Extra compiler flags, newline delimited.'
-    required: false
-    default: ''
   output:
     description: 'File to write generated table to.'
     required: false
     default: 'size_table.html'
+  check_against:
+    description: 'If set, the generated table will be compared against the given file, failing if not the same.'
```
Comment (aggarw13) [inline memory_statistics/action.yml:16]:
```suggestion
    description: 'Path to existing memory estimates file to compare with the memory estimates calculated by the action. Action will fail if sizes do not match to indicate need for updating sizes.'
```

**Review by aggarw13 — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]
